packages/@vuepress/core/lib/node/ClientComputedMixin.js

```diff
@@ -54,7 +54,7 @@ module.exports = siteData => {
       for (const path in locales) {
         if (path === '/') {
           defaultLang = locales[path]
-        } else if (this.$page.path.indexOf(path) === 0) {
+        } else if (this.$page.path.indexOf(path) === 0 || (!this.$page.path && this.$route.path.indexOf(path) === 0)) {
```

Comment:
What are you trying to solve with this condition? When would `this.$page.path` not be defined?

Reply:
@dzetah ?